superset-frontend/plugins/plugin-chart-echarts/src/Sunburst/transformProps.ts

```diff
@@ -0,0 +1,353 @@
+ * under the License.
+ */
+
+import {
+  CategoricalColorNamespace,
+  getColumnLabel,
+  getMetricLabel,
+  getNumberFormatter,
+  getSequentialSchemeRegistry,
+  getTimeFormatter,
+  NumberFormats,
+  NumberFormatter,
+  t,
+} from '@superset-ui/core';
+import { EChartsCoreOption } from 'echarts';
+import { SunburstSeriesNodeItemOption } from 'echarts/types/src/chart/sunburst/SunburstSeries';
+import { CallbackDataParams } from 'echarts/types/src/util/types';
+import { OpacityEnum } from '../constants';
+import { defaultGrid, defaultTooltip } from '../defaults';
+import { formatSeriesName, getColtypesMapping } from '../utils/series';
+import { treeBuilder, TreeNode } from '../utils/treeBuilder';
+import {
+  EchartsSunburstChartProps,
+  EchartsSunburstLabelType,
+  SunburstTransformedProps,
+} from './types';
+
+export function getLinearDomain(
+  treeData: TreeNode[],
+  callback: (treeNode: TreeNode) => number,
+) {
+  let min = 0;
+  let max = 0;
+  let temp = null;
+  function traverse(tree: TreeNode[]) {
+    tree.forEach(treeNode => {
+      if (treeNode.children?.length) {
+        traverse(treeNode.children);
+      }
+      temp = callback(treeNode);
+      if (temp !== null) {
+        if (min > temp) min = temp;
+        if (max < temp) max = temp;
+      }
+    });
+  }
+  traverse(treeData);
+  return [min, max];
+}
+
+export function formatLabel({
+  params,
+  labelType,
+  numberFormatter,
+}: {
+  params: CallbackDataParams;
+  labelType: EchartsSunburstLabelType;
+  numberFormatter: NumberFormatter;
+}): string {
+  const { name = '', value } = params;
+  const formattedValue = numberFormatter(value as number);
+
+  switch (labelType) {
+    case EchartsSunburstLabelType.Key:
+      return name;
+    case EchartsSunburstLabelType.Value:
+      return formattedValue;
+    case EchartsSunburstLabelType.KeyValue:
+      return `${name}: ${formattedValue}`;
+    default:
+      return name;
+  }
+}
+
+export function formatTooltip({
+  params,
+  numberFormatter,
+  colorByCategory,
+  totalValue,
+  metricLabel,
+  secondaryMetricLabel,
+}: {
+  params: CallbackDataParams & {
+    treePathInfo: {
+      name: string;
+      dataIndex: number;
+      value: number;
+    }[];
+  };
+  numberFormatter: NumberFormatter;
+  colorByCategory: boolean;
+  totalValue: number;
+  metricLabel: string;
+  secondaryMetricLabel?: string;
+}): string {
+  const { data, treePathInfo = [] } = params;
+  treePathInfo.shift();
+  const node = data as TreeNode;
+  const formattedValue = numberFormatter(node.value);
+  const formattedSecondaryValue = numberFormatter(node.secondaryValue);
+
+  const percentFormatter = getNumberFormatter(NumberFormats.PERCENT_2_POINT);
+  const compareValuePercentage = percentFormatter(
+    node.secondaryValue / node.value,
+  );
+  const absolutePercentage = percentFormatter(node.value / totalValue);
+  const parentNode = treePathInfo[treePathInfo.length - 1];
+  const result = [
```

Comment:
Can we try to use the [callback function](https://echarts.apache.org/en/option.html#tooltip.formatter) and return an HTML element? If not, can we at least grab font styles from the theme object? The theme object is available in the chart props.

Reply:
This will be made in a follow-up PR.